### PR TITLE
Add Python 3.11 to CI, tox, trove classifers, and README

### PR DESCRIPTION
**Pull request by joshuadavidthomas**
## Description
Adds Python 3.11 to the GHA test workflows, tox, trove classifers, and the README.

## Related Issue
N/A

## Motivation and Context
Python 3.11 was released this week.

## How Has This Been Tested?
In CI on my fork.

## Screenshots (if appropriate):
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run the `pre-commit run` command to format and lint.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] I have added my name and/or github handle to `AUTHORS.rst`
- [x] I have added my change to `CHANGES.rst`
- [x] All new and existing tests passed.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [tox]
 envlist =
-    py{37,38,39,310}-dj32-{sqlite3,postgres,mysql,mariadb},
-    py{38,39,310}-dj{40,41,main}-{sqlite3,postgres,mysql,mariadb},
+    py{37,38,39,310,311}-dj32-{sqlite3,postgres,mysql,mariadb},
```
Comment (hramezani) [inline tox.ini:3]: Seems Python 3.11 support added only for Django 4.1 and will be available in next release
https://docs.djangoproject.com/en/dev/faq/install/

File: README.rst [hunk 2]
```diff
@@ -41,9 +41,9 @@ This app supports the following combinations of Django and Python:
 ==========  ========================
   Django      Python
 ==========  ========================
-3.2         3.7, 3.8, 3.9, 3.10
-4.0         3.8, 3.9, 3.10
-4.1         3.8, 3.9, 3.10
+3.2         3.7, 3.8, 3.9, 3.10, 3.11
```
Comment (hramezani) [inline README.rst:44]: I think we don't need this change here and the line after this

File: docs/index.rst [hunk 3]
```diff
@@ -41,9 +41,9 @@ This app supports the following combinations of Django and Python:
 ==========  =======================
   Django      Python
 ==========  =======================
-3.2         3.7, 3.8, 3.9
-4.0         3.8, 3.9, 3.10
-4.1         3.8, 3.9, 3.10
+3.2         3.7, 3.8, 3.9, 3.10, 3.11
```
Comment (hramezani) [inline docs/index.rst:44]: Same as above

Comment (joshuadavidthomas) [reply docs/index.rst:44]: Ah, good catch

**Review by hramezani — APPROVED** [review]
LGTM

**Pull request merged**